### PR TITLE
`waitqueue` clarifications for SGX platform

### PR DESCRIPTION
**Pull request by raoulstrackx**
The documentation of `waitqueue` functions on the `x86_64-fortanix-unknown-sgx` platform is incorrect at some places and on others missing. This PR improves upon this.

cc: @jethrogb

### CODE REVIEW

File: library/std/src/sys/sgx/waitqueue/mod.rs [hunk 1]
```diff
@@ -204,8 +212,8 @@ impl WaitQueue {
     ) -> Result<WaitGuard<'_, T>, SpinMutexGuard<'_, WaitVariable<T>>> {
         // SAFETY: lifetime of the pop() return value is limited to the map
         // closure (The closure return value is 'static). The underlying
-        // stack frame won't be freed until after the WaitGuard created below
-        // is dropped.
+        // stack frame won't be freed until after the lock on the queue is released
+        // (i.e., `guard` is dropped).
```
Comment (jethrogb) [inline library/std/src/sys/sgx/waitqueue/mod.rs:216]: `guard` or `entry_guard`?

File: library/std/src/sys/sgx/waitqueue/mod.rs [hunk 2]
```diff
@@ -148,6 +148,8 @@ impl WaitQueue {
     /// until a wakeup event.
     ///
     /// This function does not return until this thread has been awoken.
+    ///
+    /// Safety: `before_wait` must not panic
     pub fn wait<T, F: FnOnce()>(mut guard: SpinMutexGuard<'_, WaitVariable<T>>, before_wait: F) {
```
[the remark this answers is not in the document]
Comment (jethrogb) [reply library/std/src/sys/sgx/waitqueue/mod.rs:153]: Instead, can we catch a panic and rtabort instead?

[the remark this answers is not in the document]
Comment (cuviper) [reply library/std/src/sys/sgx/waitqueue/mod.rs:153]: `std::thread::PanicGuard` is an example of that.

**Comment by workingjubilee** [comment]
Seems this is waiting on response to the comments.
@rustbot author

File: library/std/src/sys/sgx/waitqueue/mod.rs [hunk 3]
```diff
@@ -148,6 +149,8 @@ impl WaitQueue {
     /// until a wakeup event.
     ///
     /// This function does not return until this thread has been awoken.
+    ///
+    /// Safety: `before_wait` must not panic
```
Comment (jethrogb) [inline library/std/src/sys/sgx/waitqueue/mod.rs:153]: No longer required for safety

**Comment by jethrogb** [comment]
LGTM

**Comment by workingjubilee** [comment]
Looks good!
@bors r+

**Pull request merged**